custom/icds_reports/reports/take_home_ration.py

```diff
@@ -23,20 +22,27 @@ def _format_infrastructure_data(data):
 
         filters = {"month": self.month, "aggregation_level": 5}
 
-        if self.loc_level == 4:
-            filters['supervisor_id'] = self.location
-            order_by = ('awc_name',)
-        elif self.loc_level == 3:
-            filters['block_id'] = self.location
-            order_by = ('supervisor_name', 'awc_name')
-        elif self.loc_level == 2:
-            filters['district_id'] = self.location
-            order_by = ('block_name', 'supervisor_name', 'awc_name')
-        elif self.loc_level == 1:
-            filters['state_id'] = self.location
-            order_by = ('district_name', 'block_name', 'supervisor_name', 'awc_name')
+        all_order_by_columns = ('state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name')
+        if not self.location:
+            order_by = all_order_by_columns
         else:
-            order_by = ('state_name', 'district_name', 'block_name', 'supervisor_name', 'awc_name')
+            location = SQLLocation.by_location_id(self.location)
+            location_type = location.location_type.name
```

Comment:
I believe that in general, we should use a location type's `code` to do consistent matching. Names are allowed to change (that may not be the case for CAS, but its good practice for HQ)